communications/mqtt/publish/publish.ino

```diff
@@ -0,0 +1,110 @@
+#include <Wire.h>
+#include <WiFi.h>
+#include <PubSubClient.h>
+#include <ArduinoJson.h>
+
+// Replace the next variables with your Wi-Fi SSID/Password
+const char* WIFI_SSID = "WIFI_SSID";
+const char* WIFI_PASSWORD = "WIFI_PASSWORD";
+char macAddress[18];
+
+// Add MQTT Broker settings
+const char* MQTT_BROKER_IP = "BROKER_IP";
+const int MQTT_PORT = 1883;
+const bool RETAINED = true;
+const int QoS = 0;  // Quality of Service for the subscriptions
+
+WiFiClient espClient;
+PubSubClient client(espClient);
+
+// Topic Variables
+String jsonTopicStr;
+String newTopicStr;
+
+
+void setup() {
+  Serial.begin(9600);  // Starts the serial communication
+  Serial.println("");
+
+  client.setServer(MQTT_BROKER_IP, MQTT_PORT);  // Connect the configured mqtt broker
+  //client.setCallback(callback);  // Prepare what to do when a message is recieved
+
+  connectToWiFiNetwork();  // Connects to the configured network
+  connectToMqttBroker();  // Connects to the configured mqtt broker
+  
+}
+
+void loop() {
+  checkConnections();
+  publishMqttData();  // Publishes to counter topic
+  publishMqttJson();  // Subscribes to json topic
+  
+  delay(7000);
```

Comment:
that is not okay, the delay is blocking the loop function. the `checkConnections` needs to be executed as fast as posible, you need to check time on every loop and only after 7 publish the data.

This is very important to fix